gravitee-am-repository/gravitee-am-repository-mongodb-api/src/main/java/io/gravitee/am/repository/mongodb/provider/impl/MongoConnectionProvider.java

```diff
@@ -37,12 +41,16 @@
  */
 // Need to name this component to end with Repository on in order to make it injectable
 // as the Repository plugin only scan beanName ending with Repository or TransactionManager
+@Slf4j
 @Component("ConnectionProviderFromRepository")
 public class MongoConnectionProvider implements ConnectionProvider<MongoClient, MongoConnectionConfiguration>, InitializingBean {
 
     @Autowired
     private RepositoriesEnvironment environment;
 
+    @Autowired
+    private MongoFactory mongoFactory;
```

Comment:
issue: As far as I can see, only one instance of MongoConnectionProvider is available for the MAPI or the GW. If we inject the MongoFactory and if we have to specify the "prefix" in a setter before calling the getObject then it may cause issue under load where one instance of IdP maybe load with the prefix rexpected by another idp...